### DISCUSSION TITLE
Cannot grant com.termux.permission.RUN_COMMAND

### DISCUSSION THREAD

**Discussion by sunarowicz** [termux/termux-app#2417] [Q&A] [answered]
I need to grant the com.termux.permission.RUN_COMMAND permission. I followed the instruction here: [https://github.com/termux/termux-tasker](url), but there is no `Additional permissions` category in the Tasker app settings and hence no `Run commands in Termux environment` setting. I tried to grant the permission via `adb shell pm grant com.termux.permission.RUN_COMMAND`, but got the `Error: no permission specified` message. What might be wrong and how to solve it? I did not find similar problem discussed anywhere.

The phone is POCO F3 running MIUI 12.5 & rooted.

Thank you for any advice.

**Comment by agnostic-apollo** [comment, marked as answer]
What tasker version are you on?

>adb shell pm grant com.termux.permission.RUN_COMMAND

Where's the tasker package name?

**Reply by sunarowicz** [reply]
Yeah, you are right. I forgot to specify tasker package name.  The complete command `adb.exe shell pm grant net.dinglisch.android.taskerm com.termux.permission.RUN_COMMAND` worked fine.

Thank you a lot!

**Reply by agnostic-apollo** [reply]
Welcome. You need to look into android settings -> Apps -> Tasker -> Permissions -> Additional Permissions. If granting worked, then you are looking in the wrong place. Android should show it. Settings are not the one inside Tasker app itself.

**Reply by sunarowicz** [reply]
Believe me or not, I do not have any `Additional Permissions` category, nor any possibility to enable `com.termux.permission.RUN_COMMAND` permission for Tasker app in the Android system settings. Even not after enabling it via adb. I checked it carefully. And it is really enabled now, because Tasker allows me to use its Termux plugin now. But never mind, I'm happy to be able to enable the permission in the adb way.

**Reply by billyg3** [reply]
My adb just says unknown permission com.termux.permission.RUN_COMMAND

**Reply by agnostic-apollo** [reply]
Install Termux from F-Droid or Github.

**Reply by billyg3** [reply]
I had done it from apkpure does that cause problems

**Reply by agnostic-apollo** [reply]
apkpure one is from playstore, that is a different app than F-Droid/Github. We do not support that.

Read https://github.com/termux/termux-app#installation and #4000

**Reply by billyg3** [reply]
Ok I'll try that

**Comment by RobertNissan** [comment]
aquí lo tienes todo:
https://robertnissan.blogspot.com/2021/06/script-para-buscar-palabras-en-archivos.html?m=1
